pleskdistup/actions/common_checks.py

```diff
@@ -142,3 +144,109 @@ def __init__(self):
 
     def _do_check(self) -> bool:
         return subprocess.run(["/bin/fuser", "/var/lib/apt/lists/lock"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL).returncode != 0
+
+
+class MinFreeDiskSpaceViolation(typing.NamedTuple):
+    """Information about a filesystem with insufficient free disk space."""
+    dev: str
+    """Device name."""
+    req_bytes: int
+    """Required space on the device (in bytes)."""
+    avail_bytes: int
+    """Available space on the device (in bytes)."""
+    paths: typing.Set[str]
+    """Paths belonging to this device."""
+
+
+class AssertMinFreeDiskSpace(action.CheckAction):
+    """Check if there's enough free disk space.
+
+    Args:
+        requirements: A dictionary mapping paths to minimum free disk
+            space (in bytes) on the devices containing them.
+        name: Name of the check.
+        description: Description to show if the assertion is violated. If ends
+            with a colon, list of violations will be added automatically.
+    """
+    violations: typing.List[MinFreeDiskSpaceViolation]
+    """List of filesystems with insiffucient free disk space."""
+    _description: str
+
+    def __init__(
+        self,
+        requirements: typing.Dict[str, int],
+        name: str = "check if there's enough free disk space",
+        description: str = "There's not enough free disk space:",
+    ):
+        self.requirements = requirements
+        self.name = name
+        self._description = description
+        self.violations = []
+
+    @property
+    def description(self) -> str:
+        """Description of violations or empty string.
+
+        Can be set to change the message template.
+        """
+        if not self.violations:
+            return ""
+        res = self._description
+        if res.endswith(":"):
+            res += " " + ", ".join(
+                f"on filesystem {v.dev!r} for "
+                f"{', '.join(repr(p) for p in sorted(v.paths))} "
+                f"(need {v.req_bytes / 1024**2} MiB, "
+                f"got {v.avail_bytes / 1024**2} MiB)" for v in self.violations
+            )
+        return res
+
+    @description.setter
+    def description(self, val: str) -> None:
+        self._description = val
+
+    def _do_check(self) -> bool:
+        """Perform the check."""
+        log.debug("Checking minimum free disk space")
+        cmd = [
+            "findmnt", "--output", "source,target,avail",
```

Comment:
Suggest using the complete path to avoid any issues that may arise from users injecting a custom PATH. IIRC I've already meet related problems on centos2alma once.